tests/unit/test_grpc_interpreter.py

```diff
@@ -0,0 +1,193 @@
+import numpy
+
+from nidaqmx._grpc_interpreter import _assign_numpy_array
+from nidaqmx._stubs import nidaqmx_pb2
+
+
+def test___1_channel_n_sample_per_chan___update_input_array___all_samples_updated():
+    number_of_samples_per_chan = 100
+    number_of_samples_read = 100
+    initial_value = -1
+    read_value = 100
+    input_array = numpy.full(number_of_samples_per_chan, initial_value, dtype=numpy.float64)
+    output_array = numpy.full(number_of_samples_per_chan, read_value, dtype=numpy.float64)
+    response = nidaqmx_pb2.ReadAnalogF64Response(read_array=output_array)
+
+    _assign_numpy_array(
+        input_array, response.read_array, number_of_samples_read, number_of_samples_per_chan
+    )
+
+    assert (read_value == input_array).all()
+
+
+def test___1_channel_n_sample_per_chan_with_partial_data___update_input_array___only_partial_data_is_updated():
+    number_of_samples_per_chan = 100
+    number_of_samples_read = 25
+    initial_value = -1
+    read_value = 100
+    input_array = numpy.full(number_of_samples_per_chan, initial_value, dtype=numpy.float64)
+    output_array = numpy.full(number_of_samples_per_chan, read_value, dtype=numpy.float64)
+    response = nidaqmx_pb2.ReadAnalogF64Response(read_array=output_array)
+
+    _assign_numpy_array(
+        input_array, response.read_array, number_of_samples_read, number_of_samples_per_chan
+    )
+
+    assert (read_value == input_array[:number_of_samples_read]).all()
+    assert (
+        initial_value == input_array[number_of_samples_per_chan - number_of_samples_read :]
+    ).all()
```

Comment:
This will be addressed in  [Bug 2420742](https://dev.azure.com/ni/DevCentral/_workitems/edit/2420742): gRPC interpreter doesn't retain the initial array values when converting flattened array into numpy array.